packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -162,25 +162,42 @@ export default class Proxy {
       app.getApplianceRegistration(),
     ])
     const date = new Date()
+    const xenstoreData = {
+      'vm-data/system-account-xoa-password': password,
+      'vm-data/xo-proxy-authenticationToken': JSON.stringify(
+        proxyAuthenticationToken
+      ),
+      'vm-data/xoa-updater-credentials': JSON.stringify({
+        email,
+        registrationToken,
+      }),
+      'vm-data/xoa-updater-channel': JSON.stringify(xoProxyConf.channel),
+    }
+    if (networkConfiguration !== undefined) {
+      const {
+        dns = '8.8.8.8',
+        gateway,
+        ip,
+        netmask = '255.255.255.0',
+      } = networkConfiguration
+      xenstoreData['vm-data/ip'] = ip
+      xenstoreData['vm-data/gateway'] = gateway
+      xenstoreData['vm-data/netmask'] = netmask
+      xenstoreData['vm-data/dns'] = dns
+    }
     await Promise.all([
       vm.add_tags(xoProxyConf.vmTag),
       vm.set_name_label(this._generateDefaultVmName(date)),
-      vm.update_xenstore_data({
-        'vm-data/system-account-xoa-password': password,
-        'vm-data/xo-proxy-authenticationToken': JSON.stringify(
-          proxyAuthenticationToken
-        ),
-        'vm-data/xoa-updater-credentials': JSON.stringify({
-          email,
-          registrationToken,
-        }),
-        'vm-data/xoa-updater-channel': JSON.stringify(xoProxyConf.channel),
-      }),
+      vm.update_xenstore_data(xenstoreData),
     ])
 
     await xapi.startVm(vm.$id)
 
     await vm.update_xenstore_data({
+      'vm-data/dns': null,
+      'vm-data/gateway': null,
+      'vm-data/ip': null,
+      'vm-data/netmask': null,
```

Comment:
Why not using `mapValues(xenStoreData, _ => null)`?

This would simplify the code and make it less likely to forget something?